lib/pure/math.nim

```diff
@@ -156,6 +158,29 @@ func isNaN*(x: SomeFloat): bool {.inline, since: (1,5,1).} =
     when defined(js): fn()
     else: result = c_isnan(x)
 
+when defined(js):
+  proc toBitsImpl(x: float): array[2, uint32] =
+    asm """
+    const buffer = new ArrayBuffer(8);
+    const floatBuffer = new Float64Array(buffer);
+    const uintBuffer = new Uint32Array(buffer);
+    floatBuffer[0] = `x`;
+    `result` = uintBuffer
+    """
+
+proc signbit*(x: SomeFloat): bool {.inline, since: (1, 5, 1).} =
+  ## Returns true if `x` is negative, false otherwise.
+  runnableExamples:
+    doAssert not signbit(0.0)
+    doAssert signbit(-0.0)
+    doAssert signbit(-0.1)
+    doAssert not signbit(0.1)
+  when defined(js):
+    let uintBuffer = toBitsImpl(x)
+    result = (uintBuffer[1] shr 31) != 0
+  else:
+    result = c_signbit(x) != 0
```

Comment:
Would cast[bool] work to avoid branching?

Reply:
I'm pretty sure that there is no branch involved for the code as it is written for any backend that we care about.

Reply:
the codegen and asm is different:

![image](https://user-images.githubusercontent.com/2194784/103698054-a391fc00-4f55-11eb-9e94-4ad0bf8a8964.png)

but seems the same in -d:danger, so i guess it's fine